_posts/2018-08-31-announcing-riff-0-1-2-on-Knative.md

```diff
@@ -0,0 +1,199 @@
+
+#### install dockerhub push credentials and initialize the default namespace
+```
+kubectl apply -f dockerhub-push-credentials.yaml
+riff namespace init default --secret push-credentials
+```
+
+
+## feedback during builds
+There are two new options to provide feedback during builds. In previous releases, `riff function create` would return immediately after creating the build resources, without waiting for the build to succeed or fail.
+
+- `--wait` or `-w` waits until the build status is known
+- `--verbose` or `-v` is like wait, but also relays logs to the output
+
+For example, if you have not [initialized](/docs/getting-started-with-knative-riff-on-minikube/#initialize-the-namespace) the default namespace, creating a function with `--wait` will produce an error message.
+
+#### create square and push image to docker
+```
+riff function create node square \
+  --git-repo https://github.com/trisberg/node-fun-square.git \
+  --artifact square.js \
+  --image $DOCKER_ID/node-fun-square:v1 \
+  --wait
+```
+```
+Error: function creation failed: RevisionMissing: Configuration "square" does not have any ready Revision.; Revision creation failed with message: "Internal error occurred: admission webhook \"webhook.build.knative.dev\" denied the request: mutation failed: serviceaccounts \"riff-build\" not found".
+```
+
+Using `--verbose` instead of `--wait` is useful in situations where the cause of an error can  be deduced from container logs. E.g. here is the log for a missing artifact in a node build. 
+
+```
+...msg="lstat /workspace/squarexxx.js: no such file or directory"
+Error: function creation failed ...
+```
+
+## function chaining over channels
+
+[Knative/eventing](https://github.com/knative/eventing/pull/325) can now relay the reponse of a function to a another channel. This can be configured through the riff CLI by specifying the output channel with `--output` or `-o` when creating a subscription.
+
+![function chain](/images/function-chain.png)
+
+In this example, we will build a chain consisting of 3 functions and 2 channels. The random function posts numbers to the square function via the numbers channel, which forwards the output of square to the squares channel, for processing by the hello function. 
+
+To keep things interesting, we'll build the 3 functions in 3 different ways.
+- **square** uses a Knative build, pulling source code from github, and pushing the image to a registry.
+- **hello** is built manually with the docker CLI and a Dockerfile.
+- **random** uses a prebuilt image pulled from dockerhub.
+
+## square function
+The command below runs an in-cluster build of the square function, and pushes the image to dockerhub. For gcr, replace $DOCKER_ID with your gcr.io/$GCP_PROJECT.
+
+#### create square and push image to docker
+```
+riff function create node square \
+  --git-repo https://github.com/trisberg/node-fun-square.git \
+  --artifact square.js \
+  --image $DOCKER_ID/node-fun-square:v1 \
+  --wait
+```
```

Comment:
we should note that it will take some time to download the build images and that could be a `square-00001-nxwfj                            0/1       Init:0/4 ` pod for the build for several minutes

Reply:
done